### PR TITLE
[UI Strings] Improve Select in 'Add Backend to Product' form

### PR DESCRIPTION
**Pull request by lvillen**
**What this PR does / why we need it**:
**Which issue(s) this PR fixes** 

[THREESCALE-7041 [UI Strings] Improve Select in 'Add Backend to Product' form](https://issues.redhat.com/browse/THREESCALE-7041?filter=-1)

**Verification steps** 

1. Go to Products > Integration > Backends
2. Click on Add backend
3. Click on Select a backend select button
4. Click on Create a backend

Confirm the strings at all pages matches the microcopy.

**Snapshots:**

Captura de Pantalla 2022-07-18 a las 17 57 35
Captura de Pantalla 2022-07-18 a las 17 57 45
Captura de Pantalla 2022-07-18 a las 17 58 09
Captura de Pantalla 2022-07-18 a las 17 58 22
Captura de Pantalla 2022-07-18 a las 18 07 17
Captura de Pantalla 2022-07-18 a las 18 37 06
Captura de Pantalla 2022-07-18 a las 18 37 46

### CODE REVIEW

**Review by dfennessy — APPROVED** [review]
LGTM!

File: app/javascript/src/BackendApis/components/SystemNameInput.jsx [hunk 1]
```diff
@@ -14,7 +14,7 @@ const SystemNameInput = ({ systemName, setSystemName }: Props): React.Node => (
     label="SystemName"
     validated="default"
     fieldId="backend_api_system_name"
-    helperText="Only ASCII letters, numbers, dashes and underscores are allowed."
+    helperText="Only ASCII letters, numbers, dashes, and underscores are allowed."
```
Comment (josemigallas) [inline app/javascript/src/BackendApis/components/SystemNameInput.jsx:17]: @dfennessy 
TIL https://www.grammarly.com/blog/comma-before-and/

File: app/javascript/src/BackendApis/components/BackendSelect.jsx [hunk 2]
```diff
@@ -15,10 +15,11 @@ type Props = {
   backends: Backend[],
   onCreateNewBackend: () => void,
   error?: string,
-  onSelect: (Backend | null) => void
+  searchPlaceholder?: string,
+  onSelect: (Backend | null) => void,
```
Comment (josemigallas) [inline app/javascript/src/BackendApis/components/BackendSelect.jsx:19]: Now it is the same change, just 1 line after 😂

Comment (lvillen) [reply app/javascript/src/BackendApis/components/BackendSelect.jsx:19]: Well, I'll take this into account for the next time if that's ok :joy:

Comment (josemigallas) [inline app/javascript/src/BackendApis/components/BackendSelect.jsx:19]:
```suggestion
  searchPlaceholder?: string,
  onSelect: (Backend | null) => void
```

**Review by josemigallas — APPROVED** [review]

**Pull request merged**